tests/testthat/test-derive_params_growth_age.R

```diff
@@ -1,49 +1,471 @@
-    ((13 / 8)^7 - 1) / (7 * 9)
+  expected <- as.numeric()
+
+  expect_equal(
+    filter(actual, PARAMCD %in% c("WTASDS", "WTAPCTL")) %>% pull(AVAL),
+    expected
+  )
+})
+
+## Test 8: Age unit/Metadata in months works ----
+test_that("derive_params_growth_age Test 8: Age unit/Metadata in months works", {
+  vs_data <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~VISIT, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "Study", "1001", "Cycle 10 Day 1", "M", 28.6, "months", "WEIGHT", 11.3,
+    "Study", "1002", "Screening", "F", 71, "months", "WEIGHT", 27.1,
+  )
+
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 28.5, "months", -0.3277294, 13.28990, 0.1088257,
+    "F", 70.5, "months", -1.327138401, 20.03756384, 0.1484897955,
   )
+  actual <- derive_params_growth_age(
+    dataset = vs_data,
+    by_vars = exprs(STUDYID, USUBJID, VISIT),
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "WEIGHT",
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "WTASDS"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "WTAPCTL"
+    )
+  )
+
+  vs_data_meta <- vs_data %>%
+    dplyr::full_join(meta, by = c("SEX", "AGEU")) %>%
+    mutate(tmpAgeDiff = abs(AGECUR - AGE)) %>%
+    group_by(STUDYID, USUBJID, VISIT, SEX, AGECUR, AGEU, VSTESTCD, VSSTRESN) %>%
+    arrange(STUDYID, USUBJID, tmpAgeDiff) %>%
+    slice(1)
+
+  vs_sds <- vs_data_meta %>% mutate(AVAL = (((VSSTRESN / M)^L) - 1) / (L * S))
+  vs_pctl <- vs_sds %>% mutate(AVAL = pnorm(AVAL) * 100)
+  expected <- bind_rows(vs_sds, vs_pctl) %>% pull(AVAL)
+
 
   expect_equal(
-    filter(actual, PARAMCD == "WT2AGEZ") %>% pull(AVAL),
+    filter(actual, PARAMCD %in% c("WTASDS", "WTAPCTL")) %>% pull(AVAL),
     expected
   )
 })
+
+
+## Test 8: Right skew adjustment works ----
+test_that("derive_params_growth_age Test 8: Right skew adjustment works", {
+  vs_data <- tibble::tribble(
+    ~STUDYID, ~USUBJID, ~VISIT, ~SEX, ~AGECUR, ~AGEU, ~VSTESTCD, ~VSSTRESN,
+    "Study", "1001", "Screening", "M", 289, "days", "WEIGHT", 20.4,
+  )
+
+  meta <- tibble::tribble(
+    ~SEX, ~AGE, ~AGEU, ~L, ~M, ~S,
+    "M", 289, "days", 0.0868, 9.0342, 0.10885
+  )
+
+  actual <- derive_params_growth_age(
+    vs_data,
+    by_vars = exprs(STUDYID, USUBJID, VISIT),
+    sex = SEX,
+    age = AGECUR,
+    age_unit = AGEU,
+    meta_criteria = meta,
+    parameter = VSTESTCD == "WEIGHT",
+    right_skew_correction = TRUE,
+    analysis_var = VSSTRESN,
+    set_values_to_sds = exprs(
+      PARAMCD = "WGASDS",
+      PARAM = "Weight-for-age z-score"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "WGAPCTL",
+      PARAM = "Weight-for-age percentile"
+    )
+  )
+
+  SD2pos <- (9.0342 * (1 + 2*0.0868*0.10885) ^ (1/0.0868))
+  SD3pos <- (9.0342 * (1 + 3*0.0868*0.10885) ^ (1/0.0868))
+  expected_sds <- 3 + (20.4 - SD3pos)/(SD3pos - SD2pos)
+  expected_pctl <- pnorm(expected_sds)*100
+
+  expect_equal(
```

Comment:
Hi @zdz2101 , can you remind me which email you were referring to? I searched my inbox, but didn't find an email from David on right skewness adjustment.

Reply:
Ross and Zelos - I can't tell, but be sure to do this extrapolation for both high (z >3) and low (z < (-3).

Reply:
Hi @rossfarrugia , no worries at all! I'm totally fine (and actually think it is a good idea) as long as it is more efficient to move forward. 
@zdz2101 , in terms of the test, the current test for weight looks good. Do we need additional test to make sure the adjustment does not apply to height and head circumference?
@CDC-DNPAO , David, I have one follow-up question on the adjustment: since it is brought up in the WHO guidance, does it only apply to the derivation with WHO metadata (i.e., patients < 2 years old)? Or is it generalizable to derivation with CDC metadata as well?

Reply:
thanks @Minlei0201!

I wouldn't worry to add that extra test as it's up to the user (usually dictated by our template) whether they turn this option on or off, so no need to add extra tests with it turned off for other parameters.

Good point about only applying this to the WHO metadata though. David's email said "This WHO 'adjustment' is for all z-scores (and percentiles) based on the WHO chart that involves weight, so it should also apply to weight for length, but NOT to head circumference Z or length for age Z." - @zdz2101 any clever way you can figure to only apply this for the WHO metadata given we'd never know which metadata the user passes to the function (e.g. they could give only WHO and no CDC so don't think we can hardcode the <2 yrs of age here that we use in our template)? the only thing i can see is that WHO L values are only given to 4 decimal places, whereas CDC gives more, so we could use that but it feels crude 🤷 or we have to add another column to the metadata `SOURCE` = `"CDC"` or `"WHO"`?

Reply:
@Minlei0201 @zdz2101 @rossfarrugia .  The SD32 adjustment only applies to the WHO weight-based variables.  There is no comparable SD23 adjustment for the CDC data.

Reply:
thanks for confirming David! now we just need to figure out a nice way to achieve that in our package... i'm beginning to see why you created 2 separate packages now... 😉

Reply:
@rossfarrugia Ross, honestly, it was for historical reasons. I needed an R CDC package when CDC only had a SAS program (which I had rewritten in ~2005).  A few years later, my uni-based friend asked me to create an R WHO package as a favor.